OpenRA.Game/Network/OrderManager.cs

```diff
@@ -191,8 +225,11 @@ public void Tick()
 				Connection.Send(NetFrameNumber + FramesAhead, localOrders.Select(o => o.Serialize()).ToList());
 
 			localOrders.Clear();
+			clientOrders.Clear();
+
+			OrdersForFrame();
```

Comment:
After more digging I was finally able to resolve one of my main concerns about these changes: *we appear to be completely ignoring the frame defined in the packet header, so how on earth are these orders still being processed on the right tick?*

The answer is that
https://github.com/OpenRA/OpenRA/blob/406e15c2413064f08bafcdb94d81865191f2d46b/OpenRA.Game/Network/OrderManager.cs#L93-L95
inserts an initial set of empty packets into the queue, and we then add and process one frame per tick, meaning that our processing stays synchronised with the order latency and everything (hopefully) works as it should.

I think its a bad idea to leave this implicit: we should explain and validate this assumption, throwing an exception to guard against future PRs potentially regressing this. Combining with my suggestion above to inline the processing logic:

```suggestion
			var clientOrders = new List<ClientOrder>();
			foreach (var clientId in activeClients)
			{
				// The IsReadyForNextFrame check above guarantees that all clients have sent a packet
				var frameData = pendingPackets[clientId].Dequeue();

				// Orders are synchronised by sending an initial FramesAhead set of empty packets
				// and then making sure that we enqueue and process exactly one packet for each player each tick.
				// This may change in the future, so sanity check that the orders are for the frame we expect
				// and crash early instead of risking desyncs.
				var frameNumber = BitConverter.ToInt32(frameData, 0);
				if (frameNumber != NetFrameNumber)
					throw new InvalidDataException($"Attempted to process orders from client {clientId} for frame {frameNumber} on frame {NetFrameNumber}");

				foreach (var order in frameData.ToOrderList(World))
				{
					UnitOrders.ProcessOrder(this, World, clientId, order);
					clientOrders.Add(new ClientOrder { Client = clientId, Order = order });
				}
			}
```


We can then remove the `clientOrders` field on the class and remove the `.ToList()` when giving these to `UpdateSyncReport`.